x/merkledb/history.go

```diff
@@ -93,13 +98,13 @@ func (th *trieHistory) getValueChanges(startRoot, endRoot ids.ID, start []byte,
 	// [lastEndRootChange] is the last change in the history resulting in [endRoot].
 	lastEndRootChange, ok := th.lastChanges[endRoot]
 	if !ok {
-		return nil, ErrRootIDNotPresent
+		return nil, fmt.Errorf("%w: end root %s not found", ErrInsufficientHistory, endRoot)
 	}
 
 	// [startRootChanges] is the last appearance of [startRoot]
 	startRootChanges, ok := th.lastChanges[startRoot]
 	if !ok {
-		return nil, ErrStartRootNotFound
+		return nil, fmt.Errorf("%w: start root %s not found", ErrInsufficientHistory, startRoot)
```

Comment:
It seems weird to treat a missing `endRoot` differently to a missing `startRoot`... If we hit a missing `endRoot` I wouldn't expect the range proof to work either right?

Reply:
What do you mean? We also return `ErrInsufficientHistory` if the `startRoot` isn't found

Reply:
Why would we attempt to generate a range proof if we know we don't have the state to generate it?

Reply:
Still not sure I follow. This is the point where we figure out whether we have the state to generate the range proof.

Reply:
Put a TODO to return ErrNoEndRoot (or similar named error) once we require go 1.20 and then we can check for that error and not try to get a range proof if we know we have insufficient history